### DISCUSSION TITLE
how to open save and restore client ui on phoebus-win snapshot

### DISCUSSION THREAD

**Discussion by Matt-labor** [ControlSystemStudio/phoebus#2559] [Q&A] [answered]
I've deployed the save and restore service on my server , And I'm using phoebus-4.6.10-SNAPSHOT as a windows client  to test  the save-and-restore function. But I can't find the save-and-restore application in phoebus client.  Please let me know if anything is missed here.

**Comment by georgweiss** [comment]
The save-and-restore application is under Application -> Utility. 
May I ask how you build the client? Or from where you downloaded a build?

**Reply by Matt-labor** [reply]
Thank you! I downloaded the phoebus windows client from [SNS Controls CS-Studio](https://controlssoftware.sns.ornl.gov/index.html), which seems doesn't have the save-and-restore application under Utility Menu. So do I have to build my own phoebus client?

**Reply by kasemir** [reply]
Yes, in principle just `mvn clean install` should work

**Comment by georgweiss** [comment]
The SNS build does not include the save-and-restore client app.
My advice would be to build your own Phoebus app. Default build output will include "everything".

**Reply by Matt-labor** [reply]
I tried to build my my own Phoebus app from this github repository. But it still didn't include the save-and-restore app. Is that because the ant build file build.xml file doesn't contain the save-and-restore?
![image](https://user-images.githubusercontent.com/6867946/220872247-76b4b3b7-0ca5-45bd-bcc0-4a8ce17bbf48.png)
I tried to manually append the save-and-restore app to the build file, but i got a error said no build.xml in subdirectory.
![image](https://user-images.githubusercontent.com/6867946/220872871-f0845ef6-dd74-4869-b7bf-170858878516.png)

**Comment by georgweiss** [comment, marked as answer]
I recommend building with Maven.

**Reply by Matt-labor** [reply]
Thanks. I build my own app and that works.
